server/routes/all-episode-data.js

```diff
@@ -3,49 +3,94 @@
  */
 'use strict';
 
-const request = require('request');
-const dataUrl = process.env.DATA_BUCKET + 'episodes.json';
+const request = require('request'),
+      helpers = require('./helpers'),
+      dataUrl = process.env.DATA_BUCKET + 'episodes.json',
+      rssFeed = process.env.RSS_FEED,
+      inactiveEpisodes = process.env.BAD_EPISODES.split(',');
 
 // in memory cache of all available episodes as displayed on the home page (title, air date, description, number)
-let allEpisodes = [];
-
-// in memory cache of s3/cloudfront versions for each episode JSON file (waveform + transcript)
-let episodeDataVersions = {};
+let cache;
 
 // update `allEpisodes` and `episodeDataVersions`
-const update = function(cb) {
+const update = function(globalCache, cb) {
   cb = cb || function(err, success) {console.log('update show data', err || success);};
+  cache = globalCache;
 
-  // get show data from TAL's API
-  request.get({url: dataUrl, rejectUnauthorized: false}, function(err, res, body) {
-    if (!err) {
-      try {
-        // update the value of allEpisodes
-        allEpisodes = JSON.parse(body).map((item) => {
-          return {
-            'number': item.number,
-            'description': item.description,
-            'original_air_date': item.original_air_date,
-            'title': item.title
-          };
+  // get show data
+  if (rssFeed) {
+    request.get({url: rssFeed}, function(err, resp, body) {
+      if (!err) {
+        let episodes = cache.getKey('episodes') || [];
+        console.log('AAAA',episodes);
+        helpers.parseRSS(body, episodes, function(result) {
+          if (result.err) {
+            return cb(result.err);
+          }
+          else {
+            cache.setKey('allEpisodes', result.episodes);
+            cache.setKey('allEpisodesUnfiltered', result.episodesUnfiltered);
+            cache.save(true);
+            return cb(null, 'success');
+          }
         });
-        return cb(null, 'success');
-      } catch(e) {
-        console.error('unable to parse full latest episodes from TAL', e);
       }
-    }
-    return cb('no body');
-  });
+      else {
+        return cb(err);
+      }
+    });
+  }
+  else {
+    request.get({url: dataUrl, rejectUnauthorized: false}, function(err, res, body) {
+      if (!err) {
+        try {
+          let activeEpisodes = [];
+          let episodes = cache.getKey('episodes');
+          if (Array.isArray(episodes)) {
+            activeEpisodes = episodes
+                              .filter(episode => episode.enabled === true)
+                              .map(episode => episode.value);
+          }
+          // update the value of allEpisodes
+          let unfilteredEpisodes = JSON.parse(body).map((item) => {
+            return {
+              'number': item.number,
+              'description': item.description,
+              'original_air_date': item.original_air_date,
+              'title': item.title,
+              'guid': item.number
```

Comment:
In the case of podcasts that use RSS but do not use the itunes episode/season tags, `guid` will be whatever guid is defined in the RSS (usually the URI of the episode) so it will be different from `number`. But yeah, in the case of podcasts that use RSS-defined episode/season tags, or JSON-defined episode numbers, these will be the same. I think it's unavoidable though, unless we require people to use episode/season tags if they are using RSS.